src/core/service.c

```diff
@@ -1357,10 +1462,39 @@ static int service_collect_fds(
                 rfd_names[n_fds] = NULL;
         }
 
+        if (s->open_files) {
+                size_t n_fds = rn_socket_fds + s->n_fd_store;
+
+                LIST_FOREACH(open_files, f, s->open_files) {
+                        _cleanup_close_ int fd = -EBADF;
+
+                        fd = get_open_file_fd(s, f);
+                        if (fd < 0) {
+                                if (FLAGS_SET(f->flags, OPENFILE_GRACEFUL))
+                                        continue;
+
+                                return fd;
+                        }
+
+                        if (!GREEDY_REALLOC(rfds, n_fds + 1))
+                                return -ENOMEM;
+
+                        r = strv_extend(&rfd_names, f->fdname);
+                        if (r < 0)
+                                return r;
+
+                        rfds[n_fds] = TAKE_FD(fd);
+
+                        n_fds++;
+                        rn_openfile_fds++;
+                }
+        }
```

Comment:
oh, I am only realizing this only now: you are opening the fds from PID 1. Please don#t. Please move this to the child process.

opening fds might be slow, blocking, hanging. Hence we should do this in the child so that PID 1 can continue to run.